modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/defragmentation/GridQueryIndexingDefragmentation.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.persistence.defragmentation;
+
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.IgniteLogger;
+import org.apache.ignite.internal.processors.cache.CacheGroupContext;
+import org.apache.ignite.internal.processors.cache.persistence.checkpoint.CheckpointTimeoutLock;
+import org.apache.ignite.internal.processors.cache.persistence.pagemem.PageMemoryEx;
+import org.apache.ignite.internal.util.collection.IntMap;
+
+/**
+ *
+ */
+public interface GridQueryIndexingDefragmentation {
```

Comment:
Why we can't place the `startDefragmentation` method right at `GridQueryIndexing` and remove this interface?

Reply:
What if we need more methods? GridQueryIndexing is a mess already, this is just an attempt to use code decomposition.

Reply:
I don't think we will ever need any methods for defragmentation since all our attention will be switched to Ignite 3.0. So let's simplify here as much as possible and remain only the single method :-)

Reply:
Ok